hr_timesheet_purchase_order/models/hr_timesheet_recurrence.py

```diff
@@ -0,0 +1,351 @@
+
+WEEKS = {
+    "first": 1,
+    "second": 2,
+    "third": 3,
+    "last": 4,
+}
+
+
+class HRTimeSheetRecurrence(models.Model):
+    _name = "hr.timesheet.recurrence"
+    _description = "HR TimeSheet Recurrence"
+
+    task_ids = fields.One2many("hr.employee", "recurrence_id")
+    next_recurrence_date = fields.Date()
+    recurrence_left = fields.Integer(string="Number of tasks left to create")
+
+    repeat_interval = fields.Integer(string="Repeat Every", default=1)
+    repeat_unit = fields.Selection(
+        [
+            ("day", "Days"),
+            ("week", "Weeks"),
+            ("month", "Months"),
+            ("year", "Years"),
+        ],
+        default="week",
+    )
+    repeat_type = fields.Selection(
+        [
+            ("forever", "Forever"),
+            ("until", "End Date"),
+            ("after", "Number of Repetitions"),
+        ],
+        default="forever",
+        string="Until",
+    )
+    repeat_until = fields.Date(string="End Date")
+    repeat_number = fields.Integer(string="Repetitions")
+
+    repeat_on_month = fields.Selection(
+        [
+            ("date", "Date of the Month"),
+            ("day", "Day of the Month"),
+        ]
+    )
+
+    repeat_on_year = fields.Selection(
+        [
+            ("date", "Date of the Year"),
+            ("day", "Day of the Year"),
+        ]
+    )
+
+    mon = fields.Boolean(string="Mon")
+    tue = fields.Boolean(string="Tue")
+    wed = fields.Boolean(string="Wed")
+    thu = fields.Boolean(string="Thu")
+    fri = fields.Boolean(string="Fri")
+    sat = fields.Boolean(string="Sat")
+    sun = fields.Boolean(string="Sun")
+
+    repeat_day = fields.Selection([(str(i), str(i)) for i in range(1, 32)])
+    repeat_week = fields.Selection(
+        [
+            ("first", "First"),
+            ("second", "Second"),
+            ("third", "Third"),
+            ("last", "Last"),
+        ]
+    )
+    repeat_weekday = fields.Selection(
+        [
+            ("mon", "Monday"),
+            ("tue", "Tuesday"),
+            ("wed", "Wednesday"),
+            ("thu", "Thursday"),
+            ("fri", "Friday"),
+            ("sat", "Saturday"),
+            ("sun", "Sunday"),
+        ],
+        string="Day Of The Week",
+        readonly=False,
+    )
+    repeat_month = fields.Selection(
+        [
+            ("january", "January"),
+            ("february", "February"),
+            ("march", "March"),
+            ("april", "April"),
+            ("may", "May"),
+            ("june", "June"),
+            ("july", "July"),
+            ("august", "August"),
+            ("september", "September"),
+            ("october", "October"),
+            ("november", "November"),
+            ("december", "December"),
+        ]
+    )
+
+    @api.constrains("repeat_unit", "mon", "tue", "wed", "thu", "fri", "sat", "sun")
+    def _check_recurrence_days(self):
+        for timesheet in self.filtered(lambda p: p.repeat_unit == "week"):
+            if not any(
```

Comment:
Here you could use like this:
`if not any([getattr(timesheet, attr_name) for attr_name in DAYS]):`

Reply:
okay, I'll fix it